x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -599,12 +549,12 @@ export class TaskRunner<
       params: { alertId: ruleId, spaceId },
     } = this.taskInstance;
     try {
-      const decryptedAttributes = await this.getDecryptedAttributes(ruleId, spaceId);
+      const decryptedAttributes = await getDecryptedAttributes(this.context, ruleId, spaceId);
```

Comment:
Do we need to get a new fake request and rules client again here? Could we pass the rules client we get back from `loadRule` and use it here?

Reply:
Heh, ya.  Good catch - we should have caught that when the method was originally added, but who would have caught it in this maze‽‽‽.  We only need fakeRequest to create the rulesClient, so was able to cut out a bit more ...

I've had to now use `PublicMethodsOf<RulesClient>` numerous times now, so going to create an alias of that as well named `RulesClientApi`.

Reply:
code in commit 24da47bee9194f6bbfab58c047132691ac52fa5c